### PR TITLE
docs: add deprecations to upgrading guide

### PR DESCRIPTION
**Pull request by Joibel**
### Motivation

#13735 and #13719 highlighted and made some deprecations we'd like to remove in a future release.

### Modifications

To aid in ensuring users notice this also add a brief note on them to the upgrading guide.

### Verification

Links work:
image

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -5,6 +5,15 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.6
 
+### Deprecations
+
+The following features are deprecated and will be removed in a future verison of argo-workflows:
```
Comment (tico24) [inline docs/upgrading.md:10]: nit: isn't it "Argo Workflows"?

**Review by tico24 — CHANGES_REQUESTED** [review]

File: docs/upgrading.md (hunk 1 above)
Comment (Joibel) [reply docs/upgrading.md]: Done

**Review by tico24 — APPROVED** [review]

**Review by isubasinghe — APPROVED** [review]
LGTM

File: docs/upgrading.md [hunk 2]
```diff
@@ -5,6 +5,15 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.6
 
+### Deprecations
```
Comment (agilgur5) [inline docs/upgrading.md:8]: This should similarly follow order of importance like with https://github.com/argoproj/argo-workflows/pull/13720#discussion_r1790903329 et al. Deprecations would be less important than breakage

**Pull request merged**